lib/routes/validators.js

```diff
@@ -134,40 +134,67 @@ module.exports.isValidEmailAddress = function(value) {
 }
 
 module.exports.redirectTo = function redirectTo(base) {
-  const redirectTo = isA.string().max(512)
+  const validator = isA.string().max(512)
   let hostnameRegex = null
   if (base) {
     hostnameRegex = new RegExp('(?:\\.|^)' + base.replace('.', '\\.') + '$')
   }
-  redirectTo._tests.push(
+  validator._tests.push(
     {
       func: (value, state, options) => {
         if (value !== undefined && value !== null) {
-          const normalizedValue = module.exports.isValidUrl(value, hostnameRegex)
-          if (normalizedValue) {
-            return normalizedValue
+          if (module.exports.isValidUrl(value, hostnameRegex)) {
+            return value
           }
         }
 
-        return redirectTo.createError('string.base', { value }, state, options)
+        return validator.createError('string.base', { value }, state, options)
       }
     }
   )
-  return redirectTo
+  return validator
 }
 
-module.exports.isValidUrl = function (redirect, hostnameRegex) {
-  var parsed = url.parse(redirect)
+module.exports.url = function url(options) {
+  const validator = isA.string().max(512).uri(options)
```

Comment:
I notice that in the place where we use this validator, it does `max(255)` itself. Is `512` here a touch on the generous side? Could we make this `255` and then eliminate the other call to `max`?

Reply:
I think `512` here was just copied blindly from the function above; maybe we can remove the length-limit here and let individual schemas suggest one which works better?  (e.g. I assume the `.max(255)` comes from an actual constraint on db column size.